crates/std_detect/src/detect/mod.rs

```diff
@@ -97,10 +97,27 @@ cfg_if! {
     }
 }
 
+fn detect_features() -> cache::Initializer {
+    let value = self::os::detect_features();
+    if cfg!(feature = "std_detect_env_override") {
+        if let Ok(disable) = std::env::var("RUST_STD_DETECT_UNSTABLE") {
+            let mut value = value;
+            for v in disable.split(" ") {
+                let _ = Feature::from_str(v).map(|v| value.unset(v as u32));
+            }
+            value
+        } else {
+            value
+        }
+    } else {
+        value
+    }
+}
```

Comment:
I think it would be better to do it right after initializing the cache. Ideally you'll put the logic in a separate function that's `#[inline(never)]` to keep things tight.

Reply:
I'm not sure where you'd like to put it, but sure I can spin the code out of the branch.

Reply:
I think it should happen inside this branch: https://github.com/rust-lang/stdarch/blob/master/crates/std_detect/src/detect/cache.rs#L161